### DISCUSSION TITLE
How to add the "typing" python library or any mpy stub in the ESP32-S3 firmware

### DISCUSSION THREAD

**Discussion by Type-32** [lvgl-micropython/lvgl_micropython#152] [Q&A] [answered]
As suggested in the title, how can I add the `typing` python library into my firmware? I've came across a few cases where I need to use the typing library but they don't exist on the thing.

**Comment by kdschlosser** [comment, marked as answer]
Not gonna happen. There is no reason add types to your code when running MicroPython. It would take up a very large amount of memory to implement it. It would also clash with the viper code emitter.

**Comment by Type-32** [comment]
acknowledged. thank you <3
